File: frontend-angular/src/app/models/pedido.ts
import { Endereco } from "./endereco";


export type Pedido = {
  celular: string;
  data_hora: string;
  endereco_uuid: string | null;
  endereco: Endereco;
  status: any;
  frete: number;
  itens: Array<{
    loja_uuid: string,
    pedido_uuid: string,
    produto_uuid: string,
    produto_nome: string,
    produto_descricao: string,
    subtotal: number,
    quantidade: number,
    valor: number,
    observacoes: string,
    uuid: string,
  }>;
  loja_uuid: string;
  status_uuid: string | null;
  total: number
  uuid: string;
  concluido: boolean
  comentarios: string
};

File: backend/src/domain/models/pedido.py
from pydantic import BaseModel
from src.domain.models import Status, EnderecoEntrega
from typing import Optional, List, Annotated
from .item_pedido import ItemPedido, ItemPedidoPOST
from datetime import datetime


isodatetime = Annotated[str, 'isotime']


class Pedido(BaseModel):
    __tablename__ = "pedidos"

    celular: str
    data_hora: isodatetime | datetime
    loja_uuid: str
    frete: float
    concluido: bool
    comentarios: str

    status_uuid: Optional[str] = None
    usuario_uuid: Optional[str] = None
    uuid: Optional[str] = None


class PedidoGET(BaseModel):
    __tablename__ = "pedidos"
    data_hora: isodatetime | datetime
    frete: float
    loja_uuid: str
    celular: str
    total: float
    comentarios: str
    usuario_uuid: Optional[str] = None
    uuid: Optional[str] = None
    status_uuid: Optional[str] = None
    status: Optional[Status] = None
    endereco: Optional[EnderecoEntrega] = None
    itens: List[ItemPedido] = []
    concluido: Optional[bool] = None


class PedidoPOST(BaseModel):

    celular: str
    data_hora: isodatetime
    endereco: EnderecoEntrega
    frete: float
    itens: List[ItemPedidoPOST]
    loja_uuid: str
    comentarios: str
    usuario_uuid: Optional[str] = None
    status_uuid: Optional[str] = None


class AlterarStatusPedidoPATCH(BaseModel):
    status_uuid: str
